track/config.go

```diff
@@ -6,34 +6,46 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"regexp"
-	"sort"
-	"strings"
 )
 
-var (
-	rgxFunkyChars = regexp.MustCompile(`[^a-z\s-_]+`)
-	rgxSpaces     = regexp.MustCompile(`[\s-]+`)
-)
-
-// PatternGroup holds matching patterns defined in an Exercism track configuration.
+// PatternGroup holds matching patterns defined in an Exercism track config
 type PatternGroup struct {
 	IgnorePattern   string `json:"ignore_pattern,omitempty"`
 	SolutionPattern string `json:"solution_pattern,omitempty"`
 	TestPattern     string `json:"test_pattern,omitempty"`
 }
 
-// ExerciseMetadata contains metadata about an implemented exercise.
-// It's listed in the config in the order that the exercise will be
-// delivered by the API.
-type ExerciseMetadata struct {
+// ConceptKey defines the fields associated with the full array of concepts
+type ConceptKey struct {
+	Slug  string `json:"slug"`
+	UUID  string `json:"uuid"`
+	Name  string `json:"name"`
+	Blurb string `json:"blurb"`
+}
+
+// PracticeMetadata contains metadata about an implemented practice exercise.
+// The order below determines the order delivered by the API.
+type PracticeMetadata struct {
+	Slug         string   `json:"slug"`
+	UUID         string   `json:"uuid"`
+	IsCore       bool     `json:"core"`
+	AutoApprove  bool     `json:"auto_approve,omitempty"`
+	UnlockedBy   *string  `json:"unlocked_by"`
+	Difficulty   int      `json:"difficulty"`
+	Topics       []string `json:"topics,omitempty"`
```

Comment:
The `prerequisites` field also needs to be added:

```suggestion
        Prerequisites []string `json:"prerequisites,omitempty"`
	Topics       []string `json:"topics,omitempty"`
```